src/ui/public/visualize/visualize_legend.html

```diff
@@ -18,6 +21,10 @@
 
       <div class="legend-value-container">
         <div
+          kbn-accessible-click
+          data-label="{{legendData.label}}"
+          ng-focus="highlight($event)"
+          ng-blur="unhighlight($event)"
           ng-click="showDetails = !showDetails"
```

Comment:
Once the user opens these options, it's easy to get lost in the colorpicker. I know we don't want to spend time fixing this in this PR, but I think it would be helpful to let the user close the options by hitting ESC. This way if s/he gets lost in the color picker we've provided an easy way out. We just need to extract lines 22 through 75 into their own directive, and add a keypress handler there which sets `showDetails` to false. The challenge might be setting focus back on the button once the filtering options have been closed. What do you think?

Reply:
I think the color picker is not really accessible the way it is even after that PR. But I don't think just adding <kbd>Escape</kbd> would be enough. We still would need to think about color names, proper arrow navigation, etc... So I would rather replace this in the more or less near future, by using a standardized color picker from the UI framework instead and put the a11y effort into this. It would also come with the benefit, of being able to freely assign a color.

Reply:
Yeah, I hear you... I was just thinking of it as a temporary solution, since it's so easy to get stuck inside the color picker options and not know that the only way to get out of them is to backtrack all the way back to the button and then click it again.

Reply:
Done